sql/catalyst/src/main/scala/org/apache/spark/sql/types/UserDefinedType.scala

```diff
@@ -115,7 +119,9 @@ private[sql] class PythonUserDefinedType(
   }
 
   override def equals(other: Any): Boolean = other match {
-    case that: PythonUserDefinedType => this.pyUDT.equals(that.pyUDT)
+    case that: PythonUserDefinedType => pyUDT == that.pyUDT
```

Comment:
This is the same except that it works event if``pyUDT` is null.
http://stackoverflow.com/questions/7681161/whats-the-difference-between-and-equals-in-scala